slack_user_report.py

```diff
@@ -71,6 +73,18 @@ def make_report(self, ur, us, pur, pus, uid):
         blocks += self.topten(ur, pur, uid, 'you_mentioned', "The people you mentioned the most")
         blocks += self.topten(ur, pur, uid, 'mentioned_you', "The people who mentioned you the most")
         blocks += self.topten(ur, pur, uid, 'mentions_combined', "Mention Affinity")
+        # blocks += self.firstpost(uid)
+        return blocks
+
+    def firstpost(self, uid):
+        blocks = []
+        entry = self.fp.get(uid)
+        if not entry:
+            return blocks
+        url = utils.make_url(entry['slack_cid'], entry['message_id'])
+        m = "By the way, your first-ever message (to the best of our knowledge) was {}"
```

Comment:
Is the only case where we'd be wrong about their first message is when their first message is in a DM, MPDM, or private channel?

Reply:
*Probably* .  Hopefully :)